src/Windows/Host/Client/Impl/BrokerServices/LocalStaticFileServer.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.IO;
+using Microsoft.Common.Core.Logging;
+
+namespace Microsoft.R.Host.Client.BrokerServices {
+    internal class LocalStaticFileServer : StaticFileServerBase {
+        public LocalStaticFileServer(IFileSystem fs, IActionLog log, IConsole console) : base(fs, log, console) {
+        }
+
+        public async Task<string> HandleUrlAsync(string urlStr, CancellationToken ct = default(CancellationToken)) {
+            await InitializeAsync(ct);
+
+            Log.WriteLine(LogVerbosity.Minimal, MessageCategory.General, Resources.Info_StaticFileServerStarted.FormatInvariant(Listener.Prefixes.FirstOrDefault()));
+            Console.WriteLine(Resources.Info_StaticFileServerStarted.FormatInvariant(Listener.Prefixes.FirstOrDefault()));
+
+            return GetFileServerPath(urlStr);
+        }
+
+        public override async Task HandleRequestAsync(HttpListenerContext context, CancellationToken ct) {
+            UriBuilder ub = new UriBuilder();
+            ub.Scheme = "file";
+            ub.Host = "";
+            ub.Path = context.Request.Url.LocalPath;
+            ub.Query = context.Request.Url.Query;
```

Comment:
I think you can skip the query, since it's meaningless for local paths even when present (I'm actually not sure what will happen if you specify it as non-empty - might even be an error).